lib/katello/tasks/pulp3_post_migration_check.rake

```diff
@@ -0,0 +1,29 @@
+require File.expand_path("../engine", File.dirname(__FILE__))
+
+namespace :katello do
+  desc "Runs a post Pulp3 migration check for supported content types."
+  task :pulp3_post_migration_check => :environment do
+    repository_types = Katello::Pulp3::Migration::REPOSITORY_TYPES
+
+    repository_types.each do |type|
+      repositories = Katello::Repository.with_type(type)
+        .where('version_href is NULL OR remote_href is NULL')
+      if repositories.any?
+        $stderr.print("ERROR: at least one #{type} repository record has a NULL value for remote_href or version_href\n")
+        exit 1
+      end
+
+      repositories = Katello::Repository.with_type(type)
+        .where.not(remote_href: nil)
+        .where.not(version_href: nil)
+
+      non_archived_repositories = repositories.non_archived
+      if non_archived_repositories
+        .left_outer_joins(:distribution_references)
+        .where(katello_distribution_references: { id: nil }).any?
```

Comment:
Should this line be `.where(katello_distribution_references: { href: nil }).any?`

I'm wondering because we're checking hrefs for the other two fields (version_href, remote_href).

I did notice that `id` and `href` for the `katello_distribution_references` table are required to be non-null, so I'm not sure if this check is actually necessary.

Reply:
Never mind, I see what's happening here.